### PR TITLE
added "enabled" field for notification rule

### PR DESCRIPTION
**Pull request by pawelsz-rb**
## Description of the change

Added `enabled` boolean field (true/false) for rule notification.
The change is backward compatible; disabled rule (enabled = false) will not be triggered. 

The field is set to true by default. 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers assigned
- [ ] Issue from task tracker has a link to this pull request
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: .github/workflows/analyze.yml [hunk 1]
```diff
@@ -21,11 +24,6 @@ jobs:
         # a pull request then we can checkout the head.
         fetch-depth: 2
 
-    # If this run was triggered by a pull request event, then checkout
-    # the head of the pull request instead of the merge commit.
-    - run: git checkout HEAD^2
-      if: ${{ github.event_name == 'pull_request' }}
-
```
Comment (matux) [inline .github/workflows/analyze.yml:28]: Why are these changes necessary? Incl. removing the on push branches/tags (L5-6), and adding permissions (L14-18).

**Review by matux — COMMENTED** [review]
This looks all good to me, but I'm neither knowledgeable enough in Go nor Terraform to give a confident approval. So I'll wait for @waltjones to give it a look, too.

File: .github/workflows/analyze.yml (hunk 1 above)
Comment (pawelsz-rb) [reply .github/workflows/analyze.yml:28]: just merging @waltjones changes from PR #403

**Review by waltjones — APPROVED** [review]
